### ISSUE TITLE
Compiler bug when trying to use libuv

### ISSUE THREAD

**Issue by aindigo** [zig-lang/zig#603]
When trying to compile [this code](https://github.com/aindigo/ziguv) with 4f8c26d2c605f24cdeb1a4c7154662b2552640ef ,  the following errors arise :

```
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_connect_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_connect_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_shutdown_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_shutdown_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_pipe_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_pipe_t'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_read2_cb'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/c.zig:1:19: error: compiler bug: @cImport generated invalid zig code
pub const libuv = @cImport({
                  ^
?.c:1:1: note: redefinition of 'uv_read2_cb'
TODO: remember C source location to display here
^
?.c:1:1: note: previous definition is here
TODO: remember C source location to display here
^
/home/pvd/Devel/zigtests/src/main.zig:6:16: error: no member named 'uv_loop_init' in '(C import)'
    _ = c.libuv.uv_loop_init(loop);
               ^
/home/pvd/Devel/zigtests/src/main.zig:10:16: error: no member named 'uv_loop_close' in '(C import)'
    _ = c.libuv.uv_loop_close(loop);
               ^
The following command exited with error code 1:
zig build-exe /home/pvd/Devel/zigtests/src/main.zig --cache-dir /home/pvd/Devel/zigtests/zig-cache --output /home/pvd/Devel/zigtests/zig-cache/main --name main --library uv 

Build failed. Use the following command to reproduce the failure:
./zig-cache/build zig . ./zig-cache

```

Maybe it has something to do with the fact that libuv uses a lot of 'opaque' structs to hide private members with C macro magic.

**Comment by aindigo** [comment]
```
$ cat la.h
#include <uv.h>
$ zig parsec la.h
```
gives me the following output:

https://gist.github.com/aindigo/60d47c904cdfea556bd00f39103b7940

**Comment by aindigo** [comment]
I'm still getting the error compiling that repo, can you check?

another example:
```
pvd@alphanine:~/Devel/llvm5/debug$ cat la.h 
#include <uv.h>

pvd@alphanine:~/Devel/zig/build$ ./zig version
0.1.1.5029322a
pvd@alphanine:~/Devel/llvm5/debug$ ../bin/zig parsec la.h | grep "uv_connect_t ="
pub const uv_connect_t = struct_uv_connect_s;
pub const uv_connect_t = struct_uv_connect_s;
```

uv.h has the following declarations, for typedef uv_connect_t:

```
typedef struct uv_connect_s uv_connect_t;
...
...
struct uv_connect_s {       
  UV_REQ_FIELDS             
  uv_connect_cb cb;         
  uv_stream_t* handle;      
  UV_CONNECT_PRIVATE_FIELDS 
};                          

```

**Comment by andrewrk** [comment]
talked on IRC, the former issue is fixed and what is remaining is #144
